portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,106 @@
+    units.add("miles");
+    units.add("pounds");
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    response.getWriter().println("Hello Olu!");
+    String json = convertToJsonUsingGson(units);
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+    /* Commented out to make changes to part 3
+    String comment = request.getParameter("comment");
+    long timestamp = System.currentTimeMillis();
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("comment", comment);
+    commentEntity.setProperty("timestamp", timestamp);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
+    response.sendRedirect("comments.html");*/
+  }
+
+  private String convertToJson(ArrayList<String> units) {
+    String json = "{";
+    json += "\"unit 1\": ";
+    json += "\"" + units.get(0) + "\"";
+    json += ", ";
+    json += "\"unit 2\": ";
+    json += "\"" + units.get(1) + "\"";
+    json += ", ";
+    json += "\"unit 3\": ";
+    json += "\"" + units.get(2) + "\"";
+    json += ", ";
+    json += "\"unit 4\": ";
+    json += "\"" + units.get(3) + "\"";
+    json += ", ";
+    json += "\"unit 5\": ";
+    json += "\"" + units.get(4) + "\"";
+    json += "}";
+    return json;
+  }
+
+  private String convertToJsonUsingGson(ArrayList<String> units) {
+    Gson gson = new Gson();
+    String json = gson.toJson(units);
+    return json;
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    /* Preserved code from step 4
+    String name = getParameter(request, "name-input", "");
+    String email = getParameter(request, "email-input", "");
+    String phone = getParameter(request, "phone-input", "");
+    boolean emailContact = Boolean.parseBoolean(getParameter(request, "email", "false"));
+    boolean phoneContact = Boolean.parseBoolean(getParameter(request, "phone", "false"));
+    String message= "Thank You "+name+" You have not selected how you prefer to be contacted, I will reach out to you at "+email;
+    if (emailContact) {
+      message= "Thank You "+name+" I will E-mail you at "+email;
+    }
+    if (phoneContact && !emailContact) {
+      message= "Thank You "+name+" I will text out to you at "+phone;
+    }
+    if (phoneContact && emailContact) {
+      message+= ", and I will text you at "+phone;
+    }
+    response.setContentType("text/html;");
+    response.getWriter().println(message);*/
+    String comment = request.getParameter("comment");
+    long timestamp = System.currentTimeMillis();
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("commment", comment);
+    commentEntity.setProperty("timestamp", timestamp);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
```

Comment:
Just confirming that you intended this to be part of this change. I realize that things probably got mixed up because of the git issues we ran into :-)